### PR TITLE
Make imaging request wizard autofill useful defaults when possible

### PR DESCRIPTION
**Pull request by DanielStephens**
Main changes:
**Default the patient selection**
This code was already done, but it doesn't work from the menu bar, because it relies on the focused view to get the current patient. The button on the patient page works because it is in the context of the patient.
The menu bar wizard still doesn't default the patient, this is okay with Hannes. We wouldn't be able to fix this without either a hack or a change to Tryton itself as I understand it.

**Default test selection**
This is done, we default to adding all active tests that have the code `MG`

I demoed this to Hannes (Aug 4th), his comments were:

1. Move the button up above the 'Mammography reports' section (done)
2. When completing the wizard, don't navigate away from the patient page (something I added, and removed again)
3. Fix the width of the button (done, it used to be the entire screen width)
**4. Make the button taller (I've failed to do this, no idea how, tried lots of things, I think it's not important enough)**

image

### CODE REVIEW

File: health_imaging.py [hunk 1]
```diff
@@ -637,3 +642,62 @@ class BIRADS(ModelSQL, ModelView):
...
+    @fields.depends('existing_requests_count', 'patient')
+    def on_change_with_existing_requests_description(self, _name=None):
+        if self.existing_requests_count == 0:
+            return None
+        elif self.existing_requests_count == 1:
+            return 'There is 1 outstanding imaging requests for the patient'
+        else:
+            return 'There are {} outstanding imaging requests for the patient'.format(self.existing_requests_count)
+
+    def get_existing_requests_icon(self, _name):
```
Comment (DanielStephens) [inline health_imaging.py]: remove?

Comment (narendraj9) [reply health_imaging.py]: Makes sense to remove if we can directly use the icon name. I couldn't see this method being used anywhere.

File: health_vara.py [hunk 2]
```diff
@@ -184,6 +184,17 @@ def __setup__(cls):
...
 
+        # Buttons
+        cls._buttons.update({
+            'create_imaging_request': {},
+        })
+
+    @classmethod
+    @ModelView.button
+    @ModelView.button_action('health_imaging.patient_imaging_test_request')
+    def create_imaging_request(cls, records):
```
Comment (narendraj9) [inline health_vara.py:195]: The view definition [here](https://github.com/vara-ai/health_vara/pull/19/files#diff-6f504bea9dc4dfc88acf7e7daf3c92e6f9bbd4536f400c3173baf1e685ca2f17R20) seems to use `create_imaging_request` like any other field on a model. Just curious if it is possible to create view elements (buttons, links, etc.) as fields on a model?

**Review by narendraj9 — APPROVED** [review]
:+1:

File: health_vara.py (hunk 2 above)
Comment (DanielStephens) [reply health_vara.py:195]: Not sure I completely understand, but I think this is not possible. `fields` doesn't have an option for button, and I think the reason is that the value of `fields` gets persisted, whereas there is nothing sensible to persist for buttons and links.

File: health_vara_view.xml [hunk 3]
```diff
@@ -34,6 +53,38 @@ contains the full copyright notices and license terms. -->
...
+            <field name="path">icons/mammo_new_imaging_request.svg</field>
+            <field name="sequence">5</field>
+        </record>
+
+        <record model="ir.ui.icon" id="imaging_request_icon">
+            <field name="name">gnuhealth.imaging-request</field>
+            <field name="path">icons/mammo_imaging_request.svg</field>
+        </record>
+
+        <menuitem parent="health_imaging.menu_imaging"
```
Comment (rvalentini) [inline health_vara_view.xml:78]: Looks like we are creating new menu items for the custom `request` and `result` views. Are the original ones (including `draft`) still accessible somewhere in the menu tree?

**Review by rvalentini — APPROVED** [review]
Looks good, thanks for the demo :)

File: health_vara_view.xml (hunk 3 above)
Comment (DanielStephens) [reply health_vara_view.xml:78]: These menuitems existed already in gnuhealth, but I've restated them in order to change the icons.
These are in the left hand menu, and there was never a menuitem here for `draft` states.

It relates to the bottom two here:
image

The new order icon was also replaced, but it had a unique icon, so I overwrote the icon, whereas the other two had shared icons, so I had to change the menuitems to point to distinct icons instead.

**Pull request merged**